kotlin-mbedtls-metrics/src/test/kotlin/org/opencoap/ssl/transport/metrics/micrometer/DtlsServerMetricsCallbacksTest.kt

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2022-2023 kotlin-mbedtls contributors (https://github.com/open-coap/kotlin-mbedtls)
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.opencoap.ssl.transport.metrics.micrometer
+
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry
+import org.awaitility.kotlin.await
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.opencoap.ssl.EmptyCidSupplier
+import org.opencoap.ssl.PskAuth
+import org.opencoap.ssl.RandomCidSupplier
+import org.opencoap.ssl.SslConfig
+import org.opencoap.ssl.transport.BytesPacket
+import org.opencoap.ssl.transport.DatagramChannelAdapter
+import org.opencoap.ssl.transport.DtlsServer
+import org.opencoap.ssl.transport.DtlsTransmitter
+import org.opencoap.ssl.transport.HashMapSessionStore
+import org.opencoap.ssl.transport.Packet
+import org.opencoap.ssl.transport.listen
+import java.net.InetAddress
+import java.net.InetSocketAddress
+import java.nio.ByteBuffer
+import java.nio.channels.DatagramChannel
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+import java.util.function.Consumer
+import kotlin.random.Random
+
+class DtlsServerMetricsCallbacksTest {
```

Comment:
Comment: this would have been enough to be a simple unit test without setting up DtlsServer etc.. No need to change it anymore.

Reply:
Ah, right.